src/wp-includes/rest-api/endpoints/class-wp-rest-site-health-controller.php

```
@@ -0,0 +1,244 @@
+		);
+	}
+
+	/**
+	 * Check that the site can reach the WordPress.org API.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @return array
+	 */
+	public function test_dotorg_communication() {
+		return array(
+			'data' => $this->site_health->get_test_dotorg_communication()
+		);
+	}
+
+	/**
+	 * Check that loopbacks can be performed.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @return array
+	 */
+	public function test_loopback_requests() {
+		return array(
+			'data' => $this->site_health->get_test_loopback_requests()
+		);
+	}
+
+	/**
+	 * Check if the site has debug mode enabled.
+	 *
+	 * @since 5.5.0
+	 *
+	 * @return array
+	 */
+	public function test_is_in_debug_mode() {
+		return array(
+			'data' => $this->site_health->get_test_is_in_debug_mode()
+		);
+	}
+
+	/**
+	 * Get the current directory sizes for this install.
+	 *
+	 * @since 5.5.0
+	 */
+	public function get_directory_sizes() {
+		if ( ! class_exists( 'WP_Debug_Data' ) ) {
+			require_once( ABSPATH . 'wp-admin/includes/class-wp-debug-data.php' );
+		}
+
+		$sizes_data = WP_Debug_Data::get_sizes();
+		$all_sizes  = array( 'raw' => 0 );
+
+		foreach ( $sizes_data as $name => $value ) {
+			$name = sanitize_text_field( $name );
+			$data = array();
+
+			if ( isset( $value['size'] ) ) {
+				if ( is_string( $value['size'] ) ) {
+					$data['size'] = sanitize_text_field( $value['size'] );
+				} else {
+					$data['size'] = (int) $value['size'];
+				}
+			}
+
+			if ( isset( $value['debug'] ) ) {
+				if ( is_string( $value['debug'] ) ) {
+					$data['debug'] = sanitize_text_field( $value['debug'] );
+				} else {
+					$data['debug'] = (int) $value['debug'];
+				}
+			}
+
+			if ( ! empty( $value['raw'] ) ) {
+				$data['raw'] = (int) $value['raw'];
+			}
+
+			$all_sizes[ $name ] = $data;
+		}
+
+		if ( isset( $all_sizes['total_size']['debug'] ) && 'not available' === $all_sizes['total_size']['debug'] ) {
+			return new WP_Error( 'not_available', 'not_available', $all_sizes );
```

Let's return a better error message here and set a `status` code of `500`.